XenModel/Alerts/PerfmonOptionsDefinition.cs

```diff
@@ -46,15 +46,79 @@ public class PerfmonOptionsDefinition
     {
         public const String MAIL_DESTINATION_KEY_NAME = "mail-destination";
         public const String SMTP_MAILHUB_KEY_NAME = "ssmtp-mailhub";
+        public const String MAIL_LANGUAGE_KEY_NAME = "mail-language";
+
+        private class MailLanguageList
+        {
+            private List<KeyValuePair<String, String>> _list;
+
+            public MailLanguageList(List<KeyValuePair<String, String>> initList)
+            {
+                _list = new List<KeyValuePair<String, String>>(initList);
+            }
+
+            public String CodeFromName(String name)
+            {
+                String ret = null;
+
+                foreach(KeyValuePair<String, String> pair in _list)
+                {
+                    if(pair.Value == name)
+                    {
+                        ret = pair.Key;
+                        break;
+                    }
+                }
+
+                return ret;
+            }
+
+            public String NameFromCode(String code)
+            {
+                String ret = null;
+
+                foreach (KeyValuePair<String, String> pair in _list)
+                {
+                    if (pair.Key == code)
+                    {
+                        ret = pair.Value;
+                        break;
+                    }
+                }
+
+                return ret;
+            }
+
+            public int IndexFromCode(String code)
+            {
+                for (int i = 0; i < _list.Count; i++)
+                    if (code == _list[i].Key)
+                        return i;
+                return -1;
+            }
+
+            public List<KeyValuePair<String, String>> dataSource()
+            {
+                return new List<KeyValuePair<String, String>>(_list);
+            }
+        }
 
         private static readonly char[] mailHubDelim = new char[] { ':' };
         private readonly String mailHub;
         private readonly String mailDestination;
+        private readonly String mailLanguageCode;
 
-        public PerfmonOptionsDefinition(String mailHub, String mailDestination)
+        private static MailLanguageList ml_list = new MailLanguageList(new List<KeyValuePair<String, String>>() {

```

Comment:
I think this can be simplified by using a Disctionary<string, string> directly here, instead of the MailLanguageList class.